airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -648,4 +721,28 @@ private void reportCancelledAndContinueWith(final boolean isReset, final Connect
     prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
   }
 
+  private StandardSyncOutput checkFailureSyncOutput(FailureReason.FailureOrigin origin,
```

Comment:
Sorry I missed that:
I don't think that we should be using the `StandardSyncOutput` as a return parameter here. It doesn't run a sync and it is confusing to re-use the same object here. I think that we should:
- Create a dedicated pojo for the return type of this method.
- return it instead of the `StandardSyncOutput`
- If successful, return a success value. This will avoid using a null check instead of a checking the status of the check.
- change the interface of `getFailStatus` to not take sync output but just what it needs from it.

I think that the `StandardSyncOutput` is created in order to be use in `getFailStatus` only.

Reply:
All good! Today I learned the word "POJO" 🤣 https://github.com/airbytehq/airbyte/pull/12423/commits/effe6b2f264bed4b33e169d6ec0e220bb3e68d3f has a more class-based approach.